coffeecard/CoffeeCard.WebApi/Services/TokenService.cs

```diff
@@ -36,21 +41,28 @@ public string GenerateToken(IEnumerable<Claim> claims)
         public JwtSecurityToken ReadToken(string token)
         {
             return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
+
         }
 
         /// <summary>
-        ///     Checks if the JWT token is valid (Based on its lifetime)
+        ///     Receives the serialized version of a JWT token as a string. Checks if the JWT token is valid (Based on its lifetime and if it has been used before)
         /// </summary>
-        /// <param name="token"></param>
+        /// <param name="tokenString"></param>
         /// <returns></returns>
-        public bool ValidateToken(JwtSecurityToken token)
+        public async Task<bool> ValidateToken(string tokenString)
         {
             try
             {
-                if (token.ValidTo > DateTime.UtcNow) return true;
-                return false;
+                var tokenNotUsed = false;
+                var token = ReadToken(tokenString);
+
+                var user = await _claimsUtilities.ValidateAndReturnUserFromEmailClaimAsync(token.Claims);
+                
+                if (user.Tokens.Contains(new Token(tokenString))) tokenNotUsed = true;
+
+                return token.ValidTo > DateTime.UtcNow && tokenNotUsed;
             }
-            catch (Exception)
+            catch (ArgumentException)
             {
                 return false;
```

Comment:
Maybe put a log here, to log what the exception was?
```suggestion
catch (ArgumentException e)
            {
                Log.LogError($"Unable to read toke. Exception thrown = {e}");
                return false;
```